src/PointwiseFunctions/Hydro/MassFlux.cpp

```diff
@@ -3,22 +3,40 @@
 
 #include "MassFlux.hpp"
 
+#include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
 #include "Utilities/MakeWithValue.hpp"
 
 /// \cond
 namespace hydro {
 
+template <typename DataType, size_t Dim, typename Frame>
+void mass_flux(const gsl::not_null<tnsr::I<DataType, Dim, Frame>*> result,
+               const Scalar<DataType>& rest_mass_density,
+               const tnsr::I<DataType, Dim, Frame>& spatial_velocity,
+               const Scalar<DataType>& lorentz_factor,
+               const Scalar<DataType>& lapse,
+               const tnsr::I<DataType, Dim, Frame>& shift,
+               const Scalar<DataType>& sqrt_det_spatial_metric) noexcept {
+  destructive_resize_components(result, get_size(get(rest_mass_density)));
+  for (size_t i = 0; i < Dim; ++i) {
+    result->get(i) = get(rest_mass_density) * get(lorentz_factor) *
+                     get(sqrt_det_spatial_metric) *
+                     (get(lapse) * spatial_velocity.get(i) - shift.get(i));
+  }
+}
+
 template <typename DataType, size_t Dim, typename Frame>
 tnsr::I<DataType, Dim, Frame> mass_flux(
     const Scalar<DataType>& rest_mass_density,
     const tnsr::I<DataType, Dim, Frame>& spatial_velocity,
     const Scalar<DataType>& lorentz_factor, const Scalar<DataType>& lapse,
     const tnsr::I<DataType, Dim, Frame>& shift,
     const Scalar<DataType>& sqrt_det_spatial_metric) noexcept {
-  auto result =
-      make_with_value<tnsr::I<DataType, Dim, Frame>>(rest_mass_density, 0.0);
+  tnsr::I<DataType, Dim, Frame> result{};
   for (size_t i = 0; i < Dim; ++i) {
```

Comment:
For my understanding: is there a particular reason here to maintain two implementations of the mass flux, instead of changing the "return by value" function to call the "mutating" function (like you did for the lorentz factor, enthalpy, sound speed)?

Reply:
Having both wouldn't hurt, and it's also used for testing. In terms of evolution itself, I am unsure if the overload returning by value will ever be called. I guess leaving both makes things more consistent with other compute items, but we can remove it if anyone has strong feelings against them.

Reply:
I think I may not have expressed myself clearly. I trying to understand why the math equations are written out in both functions, instead of having the "return by value" function call the "mutating" function within it, like this,
```
template <...>
tnsr::I<...> mass_flux(...)  noexcept {
  tnsr::I<...> result{};
  mass_flux(make_not_null(&result), ...);
  return result;
}
```

Reply:
OK, that was definitely not intended. Something must have gone wrong when I split the local changes into different commits. Now it should be fine. Thanks for noticing this!